scripts/merge-pr.sh

```diff
@@ -37,6 +37,29 @@ check_pr_prereq() {
   fi
 }
 
+check_sync() {
+    git fetch
+    if [ -n "$(git rev-list $BRANCH..$REMOTE/$BRANCH)" ]; then
+      warn "the local branch $BRANCH is behind $REMOTE/$BRANCH."
+      echo "Do you want to cancel or do you prefer to reset local $BRANCH to $REMOTE/$BRANCH?"
+      echo "cancel/continue"
+      read response;
+      if [ "$response" = "continue" ]; then
+          echo "do you wish to save your work on a new local branch ? (y/n)";
+          read save
+          if [ "$save" = "y" ]; then
+            git commit -a -m "work saved before reset --hard"
+            git branch my-saved-work
```

Comment:
This one only works if you don't have the branch already, it dangerously commits on `BRANCH` otherwise. 

I think we are good not making the check, or just show a warning if there are modified files, what do you think?

Reply:
I am would be leaning towards a warning, rather than creating a random branch at this point. But another option, which I kind of like, is just to run a `git stash`, that way it should always work and we don't have to worry about polluting the branches. What do you think? Also it will be good to have some other devs opinion, @rasom @jeluard

Reply:
I would go with warning and exit.
@PombeirP WDYT?

Reply:
Yes, that is the main goal (avoid going through the whole review process in the script only to find out that the branch is out-of-date). If we have that warning quickly, the rest is easy.

Reply:
I've made the change and replace "continue" by "reset" as it is more explicit